src/wp-includes/post.php

```diff
@@ -5670,52 +5668,41 @@ function get_page( $page, $output = OBJECT, $filter = 'raw' ) {
  * @return WP_Post|array|null WP_Post (or array) on success, or null on failure.
  */
 function get_page_by_path( $page_path, $output = OBJECT, $post_type = 'page' ) {
-	global $wpdb;
-
-	$last_changed = wp_cache_get_last_changed( 'posts' );
-
-	$hash      = md5( $page_path . serialize( $post_type ) );
-	$cache_key = "get_page_by_path:$hash:$last_changed";
-	$cached    = wp_cache_get( $cache_key, 'posts' );
-	if ( false !== $cached ) {
-		// Special case: '0' is a bad `$page_path`.
-		if ( '0' === $cached || 0 === $cached ) {
-			return;
-		} else {
-			return get_post( $cached, $output );
-		}
-	}
-
-	$page_path     = rawurlencode( urldecode( $page_path ) );
-	$page_path     = str_replace( '%2F', '/', $page_path );
-	$page_path     = str_replace( '%20', ' ', $page_path );
-	$parts         = explode( '/', trim( $page_path, '/' ) );
-	$parts         = array_map( 'sanitize_title_for_query', $parts );
-	$escaped_parts = esc_sql( $parts );
-
-	$in_string = "'" . implode( "','", $escaped_parts ) . "'";
+	$page_path = rawurlencode( urldecode( $page_path ) );
+	$page_path = str_replace( '%2F', '/', $page_path );
+	$page_path = str_replace( '%20', ' ', $page_path );
+	$parts     = explode( '/', trim( $page_path, '/' ) );
+	$parts     = array_map( 'sanitize_title_for_query', $parts );
 
 	if ( is_array( $post_type ) ) {
 		$post_types = $post_type;
 	} else {
 		$post_types = array( $post_type, 'attachment' );
 	}
 
-	$post_types          = esc_sql( $post_types );
-	$post_type_in_string = "'" . implode( "','", $post_types ) . "'";
-	$sql                 = "
-		SELECT ID, post_name, post_parent, post_type
-		FROM $wpdb->posts
-		WHERE post_name IN ($in_string)
-		AND post_type IN ($post_type_in_string)
-	";
```

Comment:
I just noticed this is another one of those jolly queries without an orderby.

Reply:
I do `'orderby'                => 'none',`.